### PR TITLE
Add Kawayo Keyboard

### PR DESCRIPTION
**Pull request by Cipulot**
## Description

Addition of the Kawayo keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cipulot/kawayo/readme.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+Make example for this keyboard (after setting up your build environment):
+
+    make kawayo:default
+    make kawayo:via
+
+If in Vial environment:
+
+    make kawayo:vial
+
```
Comment (zvecr) [inline keyboards/cipulot/kawayo/readme.md:19]:
```suggestion
```

File: keyboards/cipulot/kawayo/readme.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+Flashing example for this keyboard:
+
+    make kawayo:default:flash
+    make kawayo:via:flash
+
+If in Vial environment:
+
+    make kawayo:vial:flash
+
```
Comment (zvecr) [inline keyboards/cipulot/kawayo/readme.md:28]:
```suggestion
```

File: keyboards/cipulot/kawayo/chconf.h [hunk 3]
```diff
@@ -0,0 +1,25 @@
+/* Copyright 2022 Cipulot
```
Comment (zvecr) [inline keyboards/cipulot/kawayo/chconf.h:1]: This file should be able to be deleted

File: keyboards/cipulot/kawayo/config.h [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+#define STM32_HSECLK 8000000
```
Comment (zvecr) [inline keyboards/cipulot/kawayo/config.h:46]:
```suggestion
```

File: keyboards/cipulot/kawayo/halconf.h [hunk 5]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2022 Cipulot
```
Comment (zvecr) [inline keyboards/cipulot/kawayo/halconf.h:1]: This file should be deleted

File: keyboards/cipulot/kawayo/mcuconf.h [hunk 6]
```diff
@@ -0,0 +1,33 @@
+/* Copyright 2022 Cipulot
```
Comment (zvecr) [inline keyboards/cipulot/kawayo/mcuconf.h:1]: This file should be able to be deleted

File: keyboards/cipulot/kawayo/rules.mk [hunk 7]
```diff
@@ -0,0 +1,20 @@
+# MCU name
+MCU = STM32F411
+
```
Comment (zvecr) [inline keyboards/cipulot/kawayo/rules.mk:3]:
```suggestion

BOARD = GENERIC_STM32_F411XE

```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/cipulot/kawayo/chconf.h (hunk 3 above)
Comment (Cipulot) [reply keyboards/cipulot/kawayo/chconf.h:1]: I can confirm. Removing this file doesn't affect workings on the firmware with the hardware.

File: keyboards/cipulot/kawayo/rules.mk (hunk 7 above)
Comment (Cipulot) [reply keyboards/cipulot/kawayo/rules.mk:3]: Added suggested line to the file. Tested working with the board proto.

**Comment by Cipulot** [comment]
I updated the paths in the readme file for the `make` and `flash` instructions. Since the board will be in the path `keyboards/cipulot/kawayo` instead of `keyboards/kawayo`.

**Comment by Cipulot** [comment]
@zvecr requested changes have been applied. A final review to complete this RP would be great. Thanks 😊

File: keyboards/cipulot/kawayo/readme.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
+# Kawayo
+
+![Kawayo Board](https://cdn.shopify.com/s/files/1/0281/3122/9834/products/Kawayoo_1024x1024@2x.jpg?v=1650025682)
```
Comment (drashna) [inline keyboards/cipulot/kawayo/readme.md:3]: Please rehost this on Imgur or a similar image hosting service. 

Shopify URLs are notoriously fickle... and disappear all the time.  So we really want/need a long term solution.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/cipulot/kawayo/readme.md (hunk 8 above)
Comment (Cipulot) [reply keyboards/cipulot/kawayo/readme.md]: @drashna changes made. Switched to Imgur hosting.

**Comment by Cipulot** [comment]
@zvecr @drashna any more changes needed?

**Comment by Cipulot** [comment]
@zvecr @drashna Any update?

**Review by keyboard-magpie — APPROVED** [review]

**Review by zvecr — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**